grails-app/views/detection/show.gsp

```diff
@@ -99,7 +99,7 @@
                                 <tbody>
                                   <g:set var="surgery" value="${detectionInstance?.surgery}" />
                                   <tr>
-                                    <td class="rowButton"><g:link class="show" controller="surgery" action="show" id="${surgery?.id}">.</g:link></td>
+                                    <td class="rowButton"><g:if test="${surgery?.id}"><g:link class="show" controller="surgery" action="show" id="${surgery?.id}">.</g:link></g:if></td>
```

Comment:
1. Was  there *always* a bug here - i.e. before the recent on-the-fly detection validation changes?
2. Can the test be simply `if surgery` rather than `if surgery?.id`?